### ISSUE TITLE
Notifications design adjustment

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/notifications#183]
The current notifications design could do with some improvements
- Avatars and action icons are fighting for attention
- Big action icons don’t look so nice and are very abstract
- The subject text is misaligned due to the avatar being inside the text at the beginning
![notifications-old](https://user-images.githubusercontent.com/925062/46278192-b8f17200-c565-11e8-9ce9-0cc98f610d62.png)

This is a quick mockup for a design adjustment:
- Avatars are bigger and the point of focus
- Actions are overlaid, smaller, white and with shadow (standard classes `icon-white icon-shadow`)
- Text is now flush left because no avatar is inside the text
![notifications-redesign](https://user-images.githubusercontent.com/925062/46278193-b98a0880-c565-11e8-995e-eee1f92f9e9d.png)

What do you think @nextcloud/designers @nickvergessen?

**Comment by skjnldsv** [comment]
I really like it! :heart_eyes: 

It still takes a bit too much height imho.
Facebook have a nice ui on this:
![capture d ecran_2018-10-01_10-47-09](https://user-images.githubusercontent.com/14975046/46278811-952f2b80-c567-11e8-9943-490dea90d4a2.png)

**Comment by nickvergessen** [comment]
Well I think it's more important what happened, than who did it.

I'm fine with removing the avatar from the name, if that is what you want. But extracting a potential actor from the notification subject and reuse their avatar as an icon is really a bad idea.

Not all notifications have an "actor". E.g. system notifications (occ notification:generate), update notifications, first-run-wizard notifications, etc.

**Comment by nickvergessen** [comment]
But yeah the misalignment should be fixed. It's caused by
https://github.com/nextcloud/notifications/blob/e8eea16946a2bcb4aa94b785cdb2120c0cdf3f55/css/styles.scss#L162

Because otherwise the avatars dont have any space when they are in the middle of the sentence.

**Comment by pixelipo** [comment]
It looks really nice, but I have to agree with @nickvergessen 

It very much depends on the context (and the app that send notification) which is more important - person or action/app; for Talk, actor is probably more important, but for Calendar, Tasks, Deck, Admin - it's the action.
For Files it's about 50-50 :smile: to complicate things further.

Lastly, I think we should add shadow only for "popover" (floating?) items, such as modals, always-on-top menu icons, sticky headers etc. and try to avoid using them to solve other design problems.

**Comment by jancborchardt** [comment]
> Lastly, I think we should add shadow only for "popover" (floating?) items, such as modals, always-on-top menu icons, sticky headers etc. and try to avoid using them to solve other design problems.

Good point, we can adjust it by displaying the dark action icon and maybe considering the layout that @skjnldsv posted. That would also move the time to a more fitting spot.

> I'm fine with removing the avatar from the name, if that is what you want. But extracting a potential actor from the notification subject and reuse their avatar as an icon is really a bad idea.

@nickvergessen The avatar should stay there of course, that’s not what I want. ;) I don’t think it’s a bad idea, and @skjnldsv’s screenshot from Facebook shows that it’s commonly done.

> Not all notifications have an "actor". E.g. system notifications (occ notification:generate), update notifications, first-run-wizard notifications, etc.

@nickvergessen Then for those notifications we still show the action icons. We simply have 2 different designs:
- When there is an "actor", we use the design I posted above, with slight modifications (like not overlapping the action icons, but showing them in dark somewhere next to the text.
- When there is no actor, simply use the action icon as a main image.

**Comment by skjnldsv** [comment]
If no avatar let use just use the app icon then. And that's it 🤗

**Comment by bartscheerlinck** [comment]
Hi, would it be possible to somehow make the notifications more clear? Several of my users are requesting some sort of pop-up window that appears as soon as a new announcement is made. Apparently, the little red dot doesn't seem to cut it for them.
